pkg/brokerapi/service_binding.go

```diff
@@ -44,9 +44,4 @@ type CreateServiceBindingResponse struct {
 
 // Credential represents connection details, username, and password that are
 // provisioned when a consumer binds to a service instance
-type Credential struct {
-	Hostname string `json:"hostname"`
-	Port     string `json:"port"`
-	Username string `json:"username"`
-	Password string `json:"password"`
-}
+type Credential map[string]interface{}
```

Comment:
It's not a delayed JSON parse. Go will still parse the data into the types as indicated in the JSON, but it won't fail if the JSON type is not what was expected in code, because everything is an `interface{}`